app/user.go

```diff
@@ -915,26 +915,81 @@ func (a *App) SetProfileImageFromFile(userId string, file io.Reader) *model.AppE
 		return model.NewAppError("SetProfileImage", "api.user.upload_profile_user.upload_profile.app_error", nil, "", http.StatusInternalServerError)
 	}
 
+	// Invalidate user cache
 	<-a.Srv.Store.User().UpdateLastPictureUpdate(userId)
+	a.invalidateUserCacheAndPublish(userId)
 
-	a.InvalidateCacheForUser(userId)
+	return nil
+}
 
-	user, userErr := a.GetUser(userId)
-	if userErr != nil {
-		mlog.Error(fmt.Sprintf("Error in getting users profile for id=%v forcing logout", userId), mlog.String("user_id", userId))
-		return nil
+func (a *App) SetIconImage(userId string, imageData *multipart.FileHeader) *model.AppError {
+	if len(*a.Config().FileSettings.DriverName) == 0 {
```

Comment:
Looks like we duplicate this check in the app and the api -- probably because we do this elsewhere. Curious if we can come up with a good way to eliminate one of them.

Reply:
Actually in `model.Config` is defined the `FileSettings.isValid` method, which only accept as valid "local" or "amazons3", so if this that has 0 length, something is really wrong.

Reply:
Fixed it by doing:
```
	if err := a.Config().IsValid(); err != nil {
		return err
	}
```
only in the app

Reply:
err, actually looks like `a.Config().IsValid();` is checking way more than i need, switching it back to `len(*a.Config().FileSettings.DriverName) == 0`

Reply:
My point here, is "a not valid config shouldn't be ever received from `App.Config()` call, the config loaded in the platform should be always valid. Actually any load or set of something in the config is always validated so... I would just remove the check entirely.

Reply:
@jespino Gotcha. Can there cases where config is not set at all, is it worth having this check in that case or shall i remove it if there is something that can catch misconfigured FileSettings down the line?